test/core/memory_usage/memory_usage_test.cc

```diff
@@ -72,26 +79,55 @@ int main(int argc, char** argv) {
   } else {
     strcpy(root, ".");
   }
-  /* start the server */
-  Subprocess svr({absl::StrCat(root, "/memory_usage_server",
-                               gpr_subprocess_binary_extension()),
-                  "--bind", grpc_core::JoinHostPort("::", port), "--nosecure",
-                  absl::StrCat("--minstack=", absl::GetFlag(FLAGS_minstack))});
-
-  /* start the client */
-  Subprocess cli({absl::StrCat(root, "/memory_usage_client",
-                               gpr_subprocess_binary_extension()),
-                  "--target", grpc_core::JoinHostPort("127.0.0.1", port),
-                  absl::StrCat("--warmup=", absl::GetFlag(FLAGS_warmup)),
-                  absl::StrCat("--benchmark=", absl::GetFlag(FLAGS_benchmark)),
-                  absl::StrCat("--minstack=", absl::GetFlag(FLAGS_minstack))});
-
-  /* wait for completion */
-  if ((status = cli.Join()) != 0) {
-    printf("client failed with: %d", status);
-    return 1;
+
+  /* Set configurations */
+  if (absl::GetFlag(FLAGS_size) == 0) absl::SetFlag(&FLAGS_size, 50000);
+
+  struct ScenarioArgs {
+    std::vector<std::string> client;
+    std::vector<std::string> server;
+  };
+  // TODO(chennancy): add in resource quota parameter setting later
+  const std::map<std::string /*scenario*/, ScenarioArgs> scenarios = {
+      {"secure", {/*client=*/{}, /*server=*/{"--secure"}}},
+      {"resource_quota", {/*client=*/{}, /*server=*/{"--secure"}}},
+      {"minstack", {/*client=*/{"--minstack"}, /*server=*/{"--minstack"}}},
+      {"insecure", {{}, {}}},
+  };
+  auto it_scenario = scenarios.find(absl::GetFlag(FLAGS_scenario_config));
+  if (it_scenario == scenarios.end()) {
+    return 3;
```

Comment:
Please leave a comment on what this error means - that no viable scenario could be found? (and maybe a printf statement as well)

Reply:
Added a print statement saying "No scenario matching the name could be found"